packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -75,112 +74,85 @@ const useDeleteWithConfirmController = <
     props: UseDeleteWithConfirmControllerParams<RecordType, ErrorType>
 ): UseDeleteWithConfirmControllerReturn => {
     const {
-        record,
-        redirect: redirectTo = 'list',
         mutationMode,
         onClick,
-        mutationOptions = {},
```

Comment:
I see one possible regression here: with the new implementation, if I'm not mistaken, you'll override the full `mutationOptions` object if one is passed. So if, say, the user overrides only the `onError` side-effect, you will loose both the default `onError` and `onSuccess` implementations. Whereas before you would only override the default `onError` but keep the default `onSuccess`. If I'm not mistaken.